src/contract/contract_function_parameters.rs

```diff
@@ -190,10 +212,10 @@ impl ContractFunctionParameters {
     }
 
     /// Add a `bytes32` argument to the `ContractFunctionParameters`
-    pub fn add_bytes32(&mut self, val: &[u8; 32]) -> &mut Self {
+    fn add_bytes32<T: IntoBytes32>(&mut self, val: T) -> &mut Self {
```

Comment:
```suggestion
    pub fn add_bytes32<T: AsRef<[u8]>>(&mut self, val: &T) -> &mut Self {
```
Might as well take anything that can give you a slice of bytes and be able to drop the trait.

Reply:
@izik1 The issue mentioned the function should take a `String`, `&str`, `&[u8;32]`. Should I change it up from the initial issue?

Reply:
If it's specifically those types then uh, right, you'd have to write the trait like this:
```rust
mod private {
    pub trait Sealed {}
    impl Sealed for String {}
    impl Sealed for str {}
    impl Sealed for [u8; 32] {}
}

pub trait AsBytes32: Sealed {
    fn as_bytes32(&self) -> &[u8];
}

impl<T: Sealed + AsRef<[u8]>> AsBytes32 for T {
    fn as_bytes32(&self) -> &[u8] {
        self.as_ref()
    }
}
```
That way you don't have to write like, all the impls and you can add types later

Reply:
note that the impl is on `str` `String` and `[u8; 32]` not `&` those, because you can call the function with an `&T` (`&str`, `&String`, `&[u8; 32]`)

Reply:
I incorporated the `AsBytes36` trait mentioned above. `&str` seems to be more ideal than `str` since `str` size isn't known at compile time.

Reply:
you do `str` instead of `&str` because you call `add_bytes32` with an `&T` -> `&str`